### PR TITLE
workloads: set workload for conmon as well

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

/kind bug
#### What this PR does / why we need it:
before, we were only setting workload settings for the containers in a pod. This was a gap, as a user may want to have the whole pod-i.e conmon-be confined as well.

Add functionality to put conmon in the workload, configurable by the "POD" label
#### Does this PR introduce a user-facing change?

```release-note
Allow users to customize conmon's resources if a pod is in a workload.
```

### CODE REVIEW

**Comment by haircommander** [comment]
still not super happy with it, needs some tweaking of the APIs but I wante to throw it at ci

**Comment by haircommander** [comment]
I've been puzzling over whether I can come up with a cleaner API between internal/oci/oci_linux and internal/config/cgmgr and I haven't. LinuxResources can be scraped from a spec (so we can reuse the pkg/config/workloads.go function MutateCgroupGivenAnnotation), and holds all the information we may want (configuration of cgroups). if someone can think of something better I'm here to listen, but I think this is ready

File: internal/config/cgmgr/systemd.go [hunk 1]
```diff
@@ -77,18 +81,100 @@ func (*SystemdManager) MoveConmonToCgroup(cid, cgroupParent, conmonCgroup string
...
+			props = append(props, systemdDbus.Property{
+				Name:  "CPUShares",
+				Value: dbus.MakeVariant(resources.CPU.Shares),
+			})
+		}
+	}
+
 	logrus.Debugf("Running conmon under slice %s and unitName %s", cgroupParent, conmonUnitName)
-	if err := utils.RunUnderSystemdScope(pid, cgroupParent, conmonUnitName, killSignalProp, systemdDbus.PropAfter("crio.service")); err != nil {
+	if err := utils.RunUnderSystemdScope(pid, cgroupParent, conmonUnitName, props...); err != nil {
```
Comment (kolyshkin) [inline internal/config/cgmgr/systemd.go:117]: Wonder if it's possible to reuse runc systemd manager's Apply()/Set() here. It does a job similar to `utils.RunUnderSystemdScope`, can be used to set arbitrary properties (via `cgroups.SystemdProps`), and in general does a better job.

Comment (haircommander) [reply internal/config/cgmgr/systemd.go:117]: unfortunatley Apply/Set aren't yet a very portable API (from what I was able to put together). Here's what I had to do to use them for this case: https://github.com/haircommander/cri-o/commit/82b09974a31c4e9c273de980592c8f1feb65c623 . I would believe I haven't used the API correctly, but I think I'd like to port this to using runc at a later time (if ever)

**Review by mrunalp — COMMENTED** [review]
Left some comments.

File: test/workloads.bats [hunk 2]
```diff
@@ -35,7 +35,7 @@ function check_cpu_fields() {
 	config=$(runtime state "$ctr_id" | jq -r .bundle)/config.json
 
 	if [ -z "$cpushares" ]; then
-		[[ $(jq -r .linux.resources.cpu.shares < "$config") == 0 ]]
+		[[ $(jq .linux.resources.cpu.shares < "$config") == *"$cpushares"* ]]
```
[the remark this answers is not in the document]
Comment (haircommander) [reply test/workloads.bats:40]: no idea how that got there haha

File: test/workloads.bats [hunk 3]
```diff
@@ -47,6 +47,51 @@ function check_cpu_fields() {
...
+			cpuset_path="/sys/fs/cgroup/cpuset"
+		fi
+
+		found_cpuset=$(cat "$cpuset_path/pod_123-456/crio-conmon-$ctr_id/cpuset.cpus")
+		if [ -z "$cpuset" ]; then
+			[[ "$cpuset" != *"$found_cpuset"* ]]
+		else
+			[[ "$cpuset" == *"$found_cpuset"* ]]
+		fi
+		# cgroup_v2 doesn't have an analogue for cpushares
```
[the remark this answers is not in the document]
Comment (haircommander) [reply test/workloads.bats]: ah! I attempted to update

**Review by mrunalp — APPROVED** [review]
/lgtm

**Pull request merged**